src/org/pentaho/platform/dataaccess/datasource/wizard/service/impl/ConnectionService.java

```diff
@@ -290,6 +291,7 @@ public Response testConnection( DatabaseConnection connection ) throws Connectio
   @Consumes( { APPLICATION_JSON } )
   @Facet( name = "Unsupported" )
   public Response updateConnection( DatabaseConnection connection ) throws ConnectionServiceException {
+    new UtilHtmlSanitizer().sanitizeConnectionParameters( connection );
```

Comment:
let's extract sanitizer to a class property, since StringEscapeUtils is thread safe as stated in [javadocs](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringEscapeUtils.html)